src/remote_gym/remote_environment_management.py

```diff
@@ -85,111 +99,300 @@ def start_as_remote_environment(
+                shape=env.action_space.shape,
+                dtype=space_to_dtype(env.action_space),
             )
         }
 
-        self.observation_spec = {
+        observation_spec = {
             1: dm_env_rpc_pb2.TensorSpec(
                 name="observation",
-                shape=self.environment.observation_space.shape,
-                dtype=space_to_dtype(self.environment.observation_space),
+                shape=env.observation_space.shape,
+                dtype=space_to_dtype(env.observation_space),
             ),
             2: dm_env_rpc_pb2.TensorSpec(name="reward", dtype=dm_env_rpc_pb2.FLOAT),
         }
 
-        action_space_bounds = space_to_bounds(self.environment.action_space)
-        observation_space_bounds = space_to_bounds(self.environment.observation_space)
+        action_space_bounds = space_to_bounds(env.action_space)
+        observation_space_bounds = space_to_bounds(env.observation_space)
 
-        tensor_spec_utils.set_bounds(
-            self.action_spec[1], minimum=action_space_bounds[0], maximum=action_space_bounds[1]
-        )
+        tensor_spec_utils.set_bounds(action_spec[1], minimum=action_space_bounds[0], maximum=action_space_bounds[1])
 
         tensor_spec_utils.set_bounds(
-            self.observation_spec[1], minimum=observation_space_bounds[0], maximum=observation_space_bounds[1]
+            observation_spec[1], minimum=observation_space_bounds[0], maximum=observation_space_bounds[1]
         )
 
         tensor_spec_utils.set_bounds(
-            self.observation_spec[2],
-            minimum=self.environment.reward_range[0],
-            maximum=self.environment.reward_range[1],
+            observation_spec[2],
+            minimum=env.reward_range[0],
+            maximum=env.reward_range[1],
         )
 
-        if self.rendering_enabled:
-            assert self.environment.render_mode == "rgb_array", (
+        if enable_rendering:
+            assert env.render_mode == "rgb_array", (
                 "Rendering remote environments is only possible if the `render_mode` attribute "
                 "of the passed environment is 'rgb_array'."
             )
-            self.environment.reset()
-            render_shape = self.environment.render().shape
-            self.observation_spec.update(
+            env.reset()
+            render_shape = env.render().shape
+            observation_spec.update(
                 {3: dm_env_rpc_pb2.TensorSpec(name="rendering", shape=render_shape, dtype=dm_env_rpc_pb2.UINT8)}
             )
             tensor_spec_utils.set_bounds(
-                self.observation_spec[3],
+                observation_spec[3],
                 minimum=0,
                 maximum=255,
             )
 
+        # Pass action and obs layout
+        out_queue.put({"action_spec": action_spec})
+        out_queue.put({"observation_spec": observation_spec})
+
+        while True:
+            action, reset = in_queue.get()
```

Comment:
Does it make sense to do the same messaging concept here as we did for the out_queue? So having something like `get_message_from_in_queue`.

Then we could also explicitly check for closing commands instead of checking if `reset is None`.

Reply:
Sure, let's stay at None. But do you think a named messaging concept like for out_queue makes sense here as well?

Reply:
Will be added in a patch.